lib/foreman_maintain/check.rb

```diff
@@ -10,8 +10,23 @@ class Check < Executable
     class Fail < StandardError
     end
 
-    def assert(condition, error_message)
-      raise Fail, error_message unless condition
+    # run condition and mark the check as failed when not passing
+    #
+    # ==== Options
+    #
+    # * +:next_steps* - one or more procedures that can be followed to address
+    #                   the failure, will be offered to the user when running
+    #                   in interactive mode
+    def assert(condition, error_message, options = {})
+      unexpected_options = options.keys - [:next_steps]
+      unless unexpected_options.empty?
+        raise ArgumentError, "Unexpected options #{unexpected_options.inspect}"
+      end
+      unless condition
+        next_steps = Array(options.fetch(:next_steps, []))
+        self.next_steps.concat(next_steps)
```

Comment:
Can we use [`add_steps`](https://github.com/iNecas/foreman_maintain/pull/29/commits/0895f48516306b3ea225a9141b3a01e179f065c1#diff-b47081cbc34a956f34fd187f253c1bc4R50) here?